example/lib/src/presentation/ui/auth/auth_bloc.dart

```diff
@@ -1,16 +1,27 @@
+import 'dart:typed_data';
+
 import 'package:bloc/bloc.dart';
+import 'dart:math';
+import 'package:polygonid_flutter_sdk/common/domain/domain_constants.dart';
+import 'package:polygonid_flutter_sdk/common/domain/domain_logger.dart';
 import 'package:polygonid_flutter_sdk/common/domain/entities/env_entity.dart';
 import 'package:polygonid_flutter_sdk/iden3comm/domain/entities/common/iden3_message_entity.dart';
 import 'package:polygonid_flutter_sdk/sdk/polygon_id_sdk.dart';
 import 'package:polygonid_flutter_sdk_example/src/data/secure_storage.dart';
 import 'package:polygonid_flutter_sdk_example/src/presentation/ui/auth/auth_event.dart';
 import 'package:polygonid_flutter_sdk_example/src/presentation/ui/auth/auth_state.dart';
+import 'package:polygonid_flutter_sdk_example/src/presentation/ui/common/widgets/profile_radio_button.dart';
+import 'package:polygonid_flutter_sdk_example/utils/nonce_utils.dart';
 import 'package:polygonid_flutter_sdk_example/utils/secure_storage_keys.dart';
 
 class AuthBloc extends Bloc<AuthEvent, AuthState> {
   final PolygonIdSdk _polygonIdSdk;
+  static const PROFILE_NONCE_KEY = "profileNonce";
+  final NonceUtils _nonceUtils;
 
-  AuthBloc(this._polygonIdSdk) : super(const AuthState.initial()) {
+  AuthBloc(this._polygonIdSdk)
+      : _nonceUtils = NonceUtils(_polygonIdSdk),
```

Comment:
we can improve this using getIt, so we don't need to pass the SDK through the classes
or another approach could be to get the SDK instance directly in the NonceUtils class with PolygonIdSdk.I